back/src/Guide/Matomo.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+
+{- | Functions for interacting with Matomo (<https://matomo.org/>, our web analytics). -}
+module Guide.Matomo
+       ( Matomo(..)
+       , postMatomo
+       ) where
+
+import Imports
+
+import Data.IP (IP)
+import Network.HTTP.Client (httpLbs, newManager, parseRequest, setQueryString)
+import Network.HTTP.Client.TLS (tlsManagerSettings)
+import Control.Concurrent.Async (withAsync)
+
+import Guide.Api.Guider (Context (..), Guider)
+import Guide.Config (Config (..))
+import Guide.Types.Edit (Edit (..))
+
+
+-- | Request with log data to send to Matomo
+data Matomo = Matomo
+  { mIP       :: Maybe IP
+  , mUA       :: Maybe Text
+  , mReferrer :: Maybe Text
+  , mTag      :: Edit
+  } deriving (Eq, Show)
+
+-- | Notify Matomo that an edit has been made.
+postMatomo :: Matomo -> Guider ()
+postMatomo Matomo{..} = do
+    Context Config{..} _ _ <- ask
+    case (_portMatomo, _pageMatomo) of
+      (Just port, Just page) -> liftIO $ do
+        flip withAsync (\_ -> pure ()) $ do
```

Comment:
Why not just `async`?

If I'm reading the docs correctly, `withAsync` will kill the thread then the lambda returns, i. e. immediately.